src/secure_allocator.hpp

```diff
@@ -99,6 +99,10 @@ bool operator!= (const secure_allocator_t<T> &, const secure_allocator_t<U> &)
 #else
 template <typename T> struct secure_allocator_t : std::allocator<T>
 {
+    template <class U> struct rebind
```

Comment:
```suggestion
    secure_allocator_t() { }
    template<class U> secure_allocator_t(const secure_allocator_t<U>&) { }
    template <class U> struct rebind
```

Reply:
> became special

It didn't.

To rebind an allocator you need to be able to determine the rebound type, and construct an instance of the rebound type from the original type. With the missing `rebind` the rebound type is `std::allocator<U>` and that can be constructed from a `secure_allocator_t<T>` (via slicing). But that's broken.

With the new rebind member the rebound type is correct: `secure_allocator_t<U>`. But you can't construct that from `secure_allocator_t<T>`. So you need to add a converting constructor, and then you also need a user-declared default constructor.

Reply:
Thank you! Taken default constructor and conversion constructor definitions from the non-no-op variant as well. I think it is equivalent to the definition you suggested.